crates/bevy_animation/src/lib.rs

```diff
@@ -691,217 +617,269 @@ where
+            warn!(
+                "Couldn't find the animation clip for {:?}",
+                target_entity.id()
+            );
+            return;
         };
-        any_path_found = true;
-        // SAFETY: The verify_no_ancestor_player check above ensures that two animation players cannot alias
-        // any of their descendant Transforms.
-        //
-        // The system scheduler prevents any other system from mutating Transforms at the same time,
-        // so the only way this fetch can alias is if two AnimationPlayers are targeting the same bone.
-        // This can only happen if there are two or more AnimationPlayers are ancestors to the same
-        // entities. By verifying that there is no other AnimationPlayer in the ancestors of a
-        // running AnimationPlayer before animating any entity, this fetch cannot alias.
-        //
-        // This means only the AnimationPlayers closest to the root of the hierarchy will be able
-        // to run their animation. Any players in the children or descendants will log a warning
-        // and do nothing.
-        let Ok(mut transform) = (unsafe { transforms.get_unchecked(target) }) else {
-            continue;
+
+        let Some(curves) = clip.get_curves(target.id) else {
+            return;
         };
-        // SAFETY: As above, there can't be other AnimationPlayers with this target so this fetch can't alias
-        let mut morphs = unsafe { morphs.get_unchecked(target) }.ok();
+
+        self.apply_curves(curves, weight, target_entity);
+    }
+
+    fn apply_curves(&self, curves: &[VariableCurve], weight: f32, target: &mut EntityMut) {
         for curve in curves {
             // Some curves have only one keyframe used to set a transform
             if curve.keyframe_timestamps.len() == 1 {
-                match &curve.keyframes {
-                    Keyframes::Rotation(keyframes) => {
-                        transform.rotation = transform.rotation.slerp(keyframes[0], weight);
-                    }
-                    Keyframes::Translation(keyframes) => {
-                        transform.translation = transform.translation.lerp(keyframes[0], weight);
-                    }
-                    Keyframes::Scale(keyframes) => {
-                        transform.scale = transform.scale.lerp(keyframes[0], weight);
-                    }
-                    Keyframes::Weights(keyframes) => {
-                        if let Some(morphs) = &mut morphs {
-                            let target_count = morphs.weights().len();
-                            lerp_morph_weights(
-                                morphs.weights_mut(),
-                                get_keyframe(target_count, keyframes, 0).iter().copied(),
-                                weight,
-                            );
-                        }
-                    }
-                }
-                continue;
+                self.apply_single_keyframe(curve, weight, target);
```

Comment:
Looks like this is redundant to call every frame if the weight isn't changing. Eventually we should see if we can avoid calling this when the transform won't change, but that's for a future PR.

Reply:
I don't think that's true, actually, as even if *this* animation clip isn't changing weight, there could be *other* transitions that will be animated right after this that are changing weights, so we'll need to call this to "reset" the transform into its expected state so that the other transitions can be correctly applied on top of it.

Reply:
In any case I'm not sure it's worth adding optimizations here as we're about to switch to the `AnimationGraph` and then all the logic will have to change at that point.

Reply:
Yep, makes sense.